python/utils.py

```diff
@@ -7,9 +8,14 @@
 SQL_FILE_DIRECTORY = "sql"
 
 
-def load_csv_file(directory: str, filename: str) -> pd.DataFrame:
-    data = pd.read_csv(f"{directory}/{filename}")
-    data.columns = data.columns.str.lower()
+def load_data_file(filepath: str) -> pd.DataFrame:
+    file_extension = Path(filepath).suffix
+    if file_extension == ".csv":
+        data = pd.read_csv(filepath)
+    elif file_extension == ".json":
+        data = pd.read_json(filepath)
+    else:
+        raise NotImplementedError(f"Unsopported data file extension: {file_extension}")
```

Comment:
Minor note but just enjoy seeing specific error types - haven't done a lot of it with Python but as we build stuff out would love to get error handling more specific and helpful